### ISSUE TITLE
Support self-signed certificates in DEX setup

### ISSUE THREAD

**Issue by stoehdoi** [litmuschaos/litmus#3846]
I am trying to use a non-trusted TLS certificate in my litmus playground setup and am failing to get DEX authentication working due to missing configuration possibilities. 

IMO, it would be great to either allow to set --insecure-sktip-tls-verify=true in the http client or allow non-trusted root CAs to be added to the auth server to overcome this issue. I know that it is also possible to mount in CA-certs via Kubernetes secrets directly into the container filesystem but I would prefer to have one of the other options available going forward.

**Comment by nickdig** [comment]
Allowing users to pass new CAs to the litmus pods will also aid users behind a company network. As mentioned above, there are ways to inject certificates, but it would be easiest if users could simply specify their certificate and update the containers CA automatically.

**Comment by stoehdoi** [comment]
Exactly, I just recognised that the litmus-subscriber is also failing to register new chaos delegates if the ingress is using a certificate which is signed by a CA which is not publicly trusted.

**Comment by nickdig** [comment]
I added a secret as recommended in the first solution and the litmus server pod did not seem to recognize my configuration. I know this because I am still getting a "cert signed by unknown authority" error when trying to connect to the public Chaos Hub.

I attempted the second solution but was unsuccessful because the certificate chain is too large. Company certificates can be thousands of lines, and this is unmanageable in a deployment file.

**Comment by gdsoumya** [comment]
The cert is only for communication between the litmus server and the agents, so if you need custom certs for anything else that won't help. Tagging @amityt for help with chaos hub issue
